invenio_rdm_records/services/components.py

```diff
@@ -144,13 +148,75 @@ def new_version(self, identity, draft=None, record=None, **kwargs):
 class ExternalPIDsComponent(ServiceComponent):
     """Service component for pids."""
 
+    def get_provider(self, scheme):
+        """Process a given PID with its provider."""
+        provider = self.service.config.pids_provides.get(scheme)
+
+        if not provider:
+            raise ValidationError(
+                _(f"No provider configured for PID type {scheme}"),
+                field_name="pids"
+            )
+
+        return provider
+
     def create(self, identity, data=None, record=None, **kwargs):
         """Inject parsed pids to the record."""
-        record.pids = data.get('pids', {})
+        pids = data.get('pids', {})
+
+        for scheme, pid_attrs in pids.items():
+            if pid_attrs.get("provider") != PID_PROVIDER_EXTERNAL:
+                provider = self.get_provider(scheme)
+
+                # PIDS-FIXME: Do we assume it fails, or should we re-raise
+                # with ValidationErrors like AccessCompoenent (I think yes)
```

Comment:
I would indeed have ValidationErrors as the other cmps given that we want to fail on a specific field.
If it was a generic error, than it probably makes more sense to have a generic one.
Eventually we change all of them at once.

Reply:
Adding a card to the board, it's quite self contained might be a good tasks to split